### ISSUE TITLE
quarkus-jackson should depend on  jackson-modules-java8

### ISSUE THREAD

**Issue by ncorai** [quarkusio/quarkus#20074]
### Description

Currently  `io.quarkus:quarkus-jackson` depends on `com.fasterxml.jackson.module:jackson-module-parameter-names`, which is actually deprecated.
It would make sense to depend on `com.fasterxml.jackson.module:jackson-modules-java8` instead, which additionally brings support for Java 8 Date/Time and Optional classes.

**Comment by geoand** [comment]
Actually `jackson-modules-java8` does not depend on any of the mentioned modules, It's just a Maven parent that specifies them.

**Comment by ncorai** [comment]
Let me try again:

1. `jackson-module-parameter-names` is deprecated
2. It's been replaced by `jackson-modules-java8`
3. Therefore, it makes sense that `io.quarkus:quarkus-jackson` should depend on `jackson-modules-java8`

**Comment by ncorai** [comment]
On page https://github.com/FasterXML/jackson-module-parameter-names it states:
"NOTE: This module has become part of Jackson Java 8 Modules as of Jackson 2.8.5
This repo still exists to allow release of patch versions of older versions; it will be hidden (made private) in near future."

**Comment by geoand** [comment]
The repositories have been deprecated, however the Maven artifacts have
not. Their source has just moved to the new repository you mention

**Comment by ncorai** [comment]
OK, I never thought of opening that reference to see it was only a pom.xml, I saw the code changed repository and I had wrongly assumed the artifacts had followed suit. Thanks for pointing that out

**Comment by geoand** [comment]
No problem 👍.

It is indeed confusing... But with Jackson 3 (whenever that arrives), we won't have all that anymore 🙂
